WalletWasabi.Gui/Controls/WalletExplorer/ReceiveTabViewModel.cs

```diff
@@ -129,6 +134,34 @@ public ReceiveTabViewModel(WalletViewModel walletViewModel)
 				SelectedAddress.InEditMode = true;
 			});
 
+			SaveQRCodeCommand = ReactiveCommand.CreateFromTask(async () =>
+			{
+				var sfd = new SaveFileDialog();
+
+				sfd.InitialFileName = $"{SelectedAddress.Address}.png";
+				sfd.InitialDirectory = Environment.GetFolderPath(Environment.SpecialFolder.MyPictures);
+				sfd.Filters.Add(new FileDialogFilter() { Name = "Portable Network Graphics (PNG) Image file", Extensions = { "png" } });
+
+				var fileFullName = await sfd.ShowAsync(Application.Current.MainWindow, fallBack: true);
+
+				if (!string.IsNullOrWhiteSpace(fileFullName))
+				{
+					var ext = Path.GetExtension(fileFullName);
+
+					if (string.IsNullOrWhiteSpace(ext))
+					{
+						fileFullName = $"{fileFullName}.png";
+					}
+					var outputStream = File.OpenWrite(fileFullName);
+
+					SelectedAddress.AddressQRCodeBitmap.Save(outputStream);
+
+					await outputStream.FlushAsync();
+
+					outputStream.Close();
+				}
+			});
```

Comment:
reactive commands should have minimal logic. Disposables aren't disposed properly.